lib/util/zip.ex

```diff
@@ -0,0 +1,135 @@
+# Copyright(c) 2015-2019 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule Antikythera.Zip do
+  @moduledoc """
+  Wrapper module for `zip` command.
+
+  For the consistency in working with antikythera and other gears, scope of this module is limited under a temporary directory reserved by `Antikythera.Tmpdir.make/2`.
+
+  Functions only accept absolute paths for both source and resulting archive.
+  """
+
+  alias Croma.Result, as: R
+  alias Antikythera.Context
+  alias Antikythera.ExecutorPool.Id, as: EPoolId
+  alias AntikytheraCore.TmpdirTracker
+
+  @typep opts :: {:encryption, boolean} | {:password, String.t}
+
+  defmodule FileName do
+    use Croma.SubtypeOfString, pattern: ~r/[^(\/\.{0,2})]\z/
+  end
+
+  @doc """
+  Creates a ZIP file.
+
+  Encryption using a password is supported in which `encryption` option is set `true`.
+
+  ## Example
+    Tmpdir.make(context, fn tmpdir ->
+      src_path = tmpdir <> "/src.txt"
+      zip_path = tmpdir <> "/archive.zip"
+      File.write!(src_path, "text")
+      Antikythera.Zip.zip(context, zip_path, src_path, [encryption: true, password: "password"])
+      |> case do
+        {:ok, archive_path} ->
+          ...
+      end
+      ...
+    end)
+  """
+  defun zip(
+    context_or_epool_id :: v[EPoolId.t | Context.t],
+    zip_raw_path        :: v[FileName.t],
+    src_raw_path        :: v[String.t],
+    opts                :: v[list(opts)] \\ []
+  ) :: R.t(Path.t) do
+    epool_id = extract_epool_id(context_or_epool_id)
+    with(
+      {:ok, tmpdir} <- TmpdirTracker.get(epool_id),
```

Comment:
If `zip/4` is called without creating `Tmpdir`, this function returns following.
```
{:error,
 {:not_found,
  %{
    state: %AntikytheraCore.TmpdirTracker.State{
      gear_tmp_dir: "/Users/takahashimasaki/src/antikythera/tmp/66687/gear_tmp",
      map: %{}
    }
  }}}
```
`state` is implementation detail, so it should not be returned.
Returing only error reason(ex.{:error, :not_found}) is better.

Reply:
Removed `state` in 89fa59c .